utils/network.go

```diff
@@ -84,14 +84,12 @@ func (n Network) URL() string {
 
 func (n Network) ChainID() *felt.Felt {
 	switch n {
-	case GOERLI:
+	case GOERLI, INTEGRATION:
 		return new(felt.Felt).SetBytes([]byte("SN_GOERLI"))
 	case MAINNET:
 		return new(felt.Felt).SetBytes([]byte("SN_MAIN"))
 	case GOERLI2:
 		return new(felt.Felt).SetBytes([]byte("SN_GOERLI2"))
-	case INTEGRATION:
```

Comment:
It is a separate network with a separate gateway URL https://github.com/NethermindEth/juno/pull/770/files/0ea2f4cf4ef97a682eadcbe1d338738d3dd58f56#diff-d51a63598746a97db8ac7bdc172764a30272c9865097535ba6aa0014751fd4d3L78

Reply:
they only share the same ChainID

Reply:
How can we verify that SN_GOERLI is valid chain_id for integration network? It is because of valid hashes while using it? I see that pathfinder use it this way, but can't find any Starknet docs related to that info.

Reply:
https://github.com/NethermindEth/juno/pull/768/commits/165d13bcf9e353c27d5a2301d2e527e2c5d258e6 proves that. Without correct ChainID transaction hashes cannot be reproduced correctly and that commit makes sure that all transaction hashes are verified for a block to be considered valid.